client/components/seo/facebook-preview/index.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { PropTypes } from 'react';
+import PureRenderMixin from 'react-pure-render/mixin';
+import classNames from 'classnames';
+import crypto from 'crypto';
+
+import {
+	firstValid,
+	hardTruncation,
+	shortEnough
+} from '../helpers';
+
+const TITLE_LENGTH = 40;
+const DESCRIPTION_LENGTH = 300;
+
+const baseDomain = url =>
+	url
+		.replace( /^[^/]+[/]*/, '' ) // strip leading protocol
+		.replace( /\/.*$/, '' ); // strip everything after the domain
+
+const blavatarUrl = slug => {
+	const slugHash = crypto
+		.createHash( 'md5' )
+		.update( slug )
+		.digest( 'hex' );
+
+	return `https://secure.gravatar.com/blavatar/${ slugHash }?s=512`;
+};
```

Comment:
Is there not a global function in Calypso somewhere to create a blavatar url? Seems like something we should have!